presto-main/src/main/java/com/facebook/presto/execution/buffer/ClientBuffer.java

```diff
@@ -98,6 +99,22 @@ public BufferInfo getInfo()
         return new BufferInfo(bufferId, destroyed, bufferedPages, sequenceId, pageBufferInfo);
     }
 
+    public List<Long> getBufferedPageBytes()
```

Comment:
We need a new method to retrieve specifically the buffered pages invidually outside of `getInfo`, as there are locking considerations described above.